safe/reportv4/report_metadata.py

```diff
@@ -0,0 +1,170 @@
+# coding=utf-8
+from importlib import import_module
+
+__author__ = 'Rizky Maulana Nugraha <lana.pcfre@gmail.com>'
+__date__ = '10/22/16'
+
+
+class ReportComponentsMetadata(object):
+
+    def __init__(
+            self, key, processor, extractor,
+            output_format, template, output_path):
+        self._key = key
+        if callable(processor):
+            self._processor = processor
+        elif isinstance(processor, basestring):
+            _package, _method = processor.rsplit('.', 1)
+            _module = import_module(_package)
+            self._processor = getattr(_module, _method)
+        self._extractor = extractor
+        self._output_format = output_format
+        self._output_path = output_path
+        self._template = template
+        self._output = None
+        self._component_context = {}
+
+    @property
```

Comment:
I know it's long to fill docstring, but it's nice to have a least `:param` and `:type` so that Pycharm can check your types for you.